driver/src/price_estimation/mod.rs

```diff
@@ -64,13 +69,13 @@ impl PriceOracle {
         unique_token_ids.into_iter().fold(
             (Vec::new(), Vec::new()),
             |(mut tokens_to_price, mut unpriced_token_ids), id| {
-                if let Some(info) = self.tokens.info(id) {
-                    tokens_to_price.push(Token {
+                match self.tokens.info(id).cloned() {
```

Comment:
Would [partition](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition) be a good use here?

Reply:
TIL! The only issue I see is that partition requires the data in both collections to be the same, here they are slightly different, one is a collection of `(TokenId, TokenInfo) <=> Token` and the other `(TokenId, Option<TokenInfo>)`.